### ISSUE TITLE
CLI parser should be case-sensitive

### ISSUE THREAD

**Issue by MitchAman** [eclipse-vertx/vert.x#2844]
When using the [CLI API](https://vertx.io/docs/vertx-core/java/#_vert_x_command_line_interface_api) for parsing command-line parameters I noticed that the _shortName_ for Options are case-insensitive.  I wouldn't have expected this behavior by default, and didn't see this behavior documented anywhere.  I recall working with other command-line tools which accept single-character ("shortName") options in both uppercase and lowercase forms (i.e. `./someTool -a someValue -A someOtherValue`).

Here's a quick test case demonstrating this behavior (short names):
```
    @Test
    public void testCliOptionShortNamesAreCaseSensitive() {
        CLI cli = CLI.create("test")
                .addOption(new Option().setShortName("a").setLongName("lowercase"))
                .addOption(new Option().setShortName("A").setLongName("uppercase"));

        String lowercaseValue = "someValue";
        String uppercaseValue = "someOtherValue";

        CommandLine commandLine = cli.parse(Arrays.asList("-a", lowercaseValue,
                                                          "-A", uppercaseValue));

        Assert.assertEquals(lowercaseValue, commandLine.getOptionValue("a"));
        Assert.assertEquals(uppercaseValue, commandLine.getOptionValue("A"));
    }
```
```
org.junit.ComparisonFailure: 
Expected :someOtherValue
Actual   :someValue
```

And here's another, where I pass the long names into `commandLine.getOptionValue()`:
```
    @Test
    public void testCliOptionLongNamesAreCaseSensitive() {
        CLI cli = CLI.create("test")
                .addOption(new Option().setShortName("a").setLongName("lowercase"))
                .addOption(new Option().setShortName("A").setLongName("uppercase"));

        String lowercaseValue = "someValue";
        String uppercaseValue = "someOtherValue";

        CommandLine commandLine = cli.parse(Arrays.asList("-a", lowercaseValue,
                                                          "-A", uppercaseValue));

        Assert.assertEquals(lowercaseValue, commandLine.getOptionValue("lowercase"));
        Assert.assertEquals(uppercaseValue, commandLine.getOptionValue("uppercase"));
    }
```
```
java.lang.AssertionError: 
Expected :someOtherValue
Actual   :null
```

Interestingly enough, if I pass the full long name representations into `cli.parse()`, and use the full long names when retrieving the values via `commandLine.getOptionValue()`, then I get the expected behavior (test passes):
```
    @Test
    public void testCliOptionLongNames() {
        CLI cli = CLI.create("test")
                .addOption(new Option().setShortName("a").setLongName("lowercase"))
                .addOption(new Option().setShortName("A").setLongName("uppercase"));

        String lowercaseValue = "someValue";
        String uppercaseValue = "someOtherValue";

        CommandLine commandLine = cli.parse(Arrays.asList("--lowercase=" + lowercaseValue,
                                                          "--uppercase=" + uppercaseValue));

        Assert.assertEquals(lowercaseValue, commandLine.getOptionValue("lowercase"));
        Assert.assertEquals(uppercaseValue, commandLine.getOptionValue("uppercase"));
    }
```

**Comment by vietj** [comment]
@MitchAman can you contribute a few tests and a fix ?

**Comment by MitchAman** [comment]
@vietj Sure, I can do that.  Once I find some time I'll read through the contributing docs on the wiki, then get going on a fix.

**Comment by tsegismont** [comment]
Fixed in #3067
